llmfoundry/callbacks/envlogger.py

```diff
@@ -0,0 +1,177 @@
+                'available': psutil.virtual_memory().available,
+            },
+        }
+
+    def _get_gpu_info(self) -> dict[str, Any]:
+        if torch.cuda.is_available():
+            return {
+                'model': torch.cuda.get_device_name(0),
+                'count': torch.cuda.device_count(),
+                'memory': {
+                    'total': torch.cuda.get_device_properties(0).total_memory,
+                    'allocated': torch.cuda.memory_allocated(0),
+                },
+            }
+        return {'available': False}
+
+    def _get_nvidia_info(self) -> dict[str, Any]:
+        if torch.cuda.is_available():
+            nccl_version = torch.cuda.nccl.version()  # type: ignore
+            return {
+                'cuda_version':
+                    torch.version.cuda,  # type: ignore[attr-defined]
+                'cudnn_version': str(torch.backends.cudnn.version()
+                                    ),  # type: ignore[attr-defined]
+                'nccl_version': '.'.join(
+                    map(str, nccl_version),
+                ),
+            }
+
+        return {'available': False}
+
+    def _get_distributed_info(self) -> dict[str, Any]:
+        return {
+            'world_size': int(os.environ.get('WORLD_SIZE', 1)),
+            'local_world_size': int(os.environ.get('LOCAL_WORLD_SIZE', 1)),
+            'rank': int(os.environ.get('RANK', 0)),
+            'local_rank': int(os.environ.get('LOCAL_RANK', 0)),
+        }
+
+    def _get_docker_info(self) -> dict[str, Any]:
+        from mcli import sdk
+
+        run = sdk.get_run(os.environ['RUN_NAME'])
+        image, tag = run.image.split(':')
+        return {
+            'image': image,
+            'tag': tag,
+        }
+
+    def fit_start(self, state: State, logger: Logger) -> None:
+        # Collect environment data
+        if self.log_git:
+            self.env_data['git_info'] = {
+                folder:
+                self._get_git_info(os.path.join(self.workspace_dir, folder))
+                for folder in os.listdir(self.workspace_dir)
+                if os.path.isdir(os.path.join(self.workspace_dir, folder))
+            }
+
+        if self.log_packages:
+            self.env_data['package_versions'] = {
+                pkg: self._get_package_version(pkg)
+                for pkg in self.packages_to_log
+            }
+        if self.log_nvidia:
+            self.env_data['nvidia'] = self._get_nvidia_info()
+
+        if self.log_docker:
+            self.env_data['docker'] = self._get_docker_info()
+        if self.log_system:
+            self.env_data['system_info'] = self._get_system_info()
+
+        if self.log_gpu:
+            self.env_data['gpu_info'] = self._get_gpu_info()
+
+        if self.log_distributed:
+            self.env_data['distributed_info'] = self._get_distributed_info()
+
+        # Log the collected data
+        logger.log_hyperparameters({'environment_data': self.env_data})
```

Comment:
Does this get logged to mosaciml logger?

Reply:
Yup, https://github.com/mosaicml/composer/blob/main/composer/loggers/mosaicml_logger.py#L94